controllers/sandbox_controller.go

```diff
@@ -0,0 +1,139 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	sandboxv1alpha1 "sigs.k8s.io/agent-sandbox/api/v1alpha1"
+)
+
+// SandboxReconciler reconciles a Sandbox object
+type SandboxReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=agents.x-k8s.io,resources=sandboxes,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=agents.x-k8s.io,resources=sandboxes/status,verbs=get;update;patch
+
+//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Sandbox object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.1/pkg/reconcile
+func (r *SandboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	sandbox := &sandboxv1alpha1.Sandbox{}
+	if err := r.Get(ctx, req.NamespacedName, sandbox); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("sandbox resource not found. Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if !sandbox.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.Info("Sandbox is being deleted")
+		return ctrl.Result{}, nil
+	}
+
+	// Check if the pod already exists, if not create a new one
+	podInCluster := &corev1.Pod{}
+	found := true
+	err := r.Get(ctx, types.NamespacedName{Name: sandbox.Name, Namespace: sandbox.Namespace}, podInCluster)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			log.Error(err, "Failed to get Pod")
+			return ctrl.Result{}, err
+		}
+		found = false
+	}
+
+	// Create a pod object from the sandbox
+	pod, err := r.podForSandbox(sandbox)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if !found {
+		log.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+		if err = r.Create(ctx, pod, client.FieldOwner("sandbox-controller")); err != nil {
+			log.Error(err, "Failed to create", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+			return ctrl.Result{}, err
+		}
+	} else {
+		log.Info("Found Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+		if err = r.Patch(ctx, pod, client.Apply, client.ForceOwnership, client.FieldOwner("sandbox-controller")); err != nil {
```

Comment:
Please remove the patch logic completely. Let's focus only on pod creation in this PR. 

Pod updates can be delete-and-recreate (i.e. all pod fields are mutable) or in-place update (i.e. only some pod fields are mutable, e.g. images, resources requests/limits). That can be added later in a follow-up PR along with related API changes.

Reply:
done.